File: backend/cloude_storage/pkg/service/service.go
package service

import (
	"stor/pkg/domain"
	"stor/pkg/repository"
)

type CloudeInterface interface {
	NewDir(data domain.DirData) bool
	NewFile(data domain.FileData) bool

	WriteFile(data domain.WriteData) bool
	ReadFile(data domain.FileData) repository.ResponeData

	DeleteFile(data domain.FileData) bool
	DeleteDir(data domain.DirData) bool

	IsFile(data domain.FileData) bool
	IsDir(data domain.DirData) bool

	DirNameAll(path string) []string
	FileNameAll(path string) []string
}

type Service struct {
	CloudeInterface
	Cloude *repository.Cloude
}

func NewService(cloude *repository.Cloude) *Service {
	return &Service{Cloude: cloude}
}

func (s *Service) NewFile(data domain.FileData) bool {
	return s.Cloude.NewFile(data)
}

func (s *Service) NewDir(data domain.DirData) bool {
	return s.Cloude.NewDir(data)
}

func (s *Service) WriteFile(data domain.WriteData) bool {
	return s.Cloude.WriteFile(data)
}

func (s *Service) ReadFile(data domain.FileData) repository.ResponeData {
	return s.Cloude.ReadFile(data)
}

func (s *Service) DeleteFile(data domain.FileData) bool {
	return s.Cloude.DeleteFile(data)
}

func (s *Service) DeleteDir(data domain.DirData) bool {
	return s.Cloude.DeleteDir(data)
}

func (s *Service) IsFile(data domain.FileData) bool {
	return s.Cloude.IsFile(data)
}

func (s *Service) IsDir(data domain.DirData) bool {
	return s.Cloude.IsDir(data)
}

func (s *Service) FileNameAll(path string) []string {
	return s.Cloude.FileNameAll(path)
}

func (s *Service) DirNameAll(path string) []string {
	return s.Cloude.DirNameAll(path)
}

File: backend/cloude_storage/src/core/service/service.cpp
#include "service.h"
#include "../filesys/filesys.h"


namespace Service {
    void DeleteDirCloude(nlohmann::json&  json){
        Filesystem::type::DirData dir = {
            json["name"],
            json["path"]
        };
        Filesystem::delete_dir(dir);
    }

    void DeleteFileCloude(nlohmann::json& json){
        Filesystem::type::FileData file = {
            json["name"],
            json["path"],
            json["type"]
        };
        Filesystem::delete_file(file);
    }

    void CreateFileCloude(nlohmann::json&  json){
        Filesystem::type::FileData file = {
            json["name"],
            json["path"],
            json["type"]
        };
        Filesystem::create_file(file);
    }

    void CreateDirCloude(nlohmann::json&  json){
        Filesystem::type::DirData dir = {
            json["name"],
            json["path"]
        };
        Filesystem::create_dir(dir);
    }

    std::string ReadFileCloude(nlohmann::json&  json){
        Filesystem::type::FileData file = {
            json["name"],
            json["path"],
            json["type"]
        };

        return Filesystem::read_file(file);
    }
    void WriteFileCloude(nlohmann::json&  json){
        Filesystem::type::FileData file = {
            json["name"],
            json["path"],
            json["type"]
        };
        std::string data = json["data"];

        Filesystem::write_file(file,data.c_str());
    }
    bool IsFileCloude(nlohmann::json&  json){
        Filesystem::type::FileData file = {
            json["name"],
            json["path"],
            json["type"]
        };
        return Filesystem::is_file(file);
    }
    bool IsDirCloude(nlohmann::json&  json){
        Filesystem::type::DirData dir = {
            json["name"],
            json["path"]
        };
        return Filesystem::is_dir(dir);       
    }
    std::vector<std::string> ListNameFileAll(nlohmann::json&  json){
        return Filesystem::list_name_file(std::string(json["path"]).c_str());
    }

    std::vector<std::string> ListNameDirAll(nlohmann::json& json){
        return Filesystem::list_name_dir(std::string(json["path"]).c_str());
    }
}